Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -41,38 +41,99 @@ public AsyncFunctionAssertions(Func<TTask> subject, IExtractExceptions extractor
-                await completedTask;
-            }
+        using var timeoutCancellationTokenSource = new CancellationTokenSource();
+        Task completedTask =
+            await Task.WhenAny(task, Clock.DelayAsync(remainingTime, timeoutCancellationTokenSource.Token));
 
-            Execute.Assertion
-                .ForCondition(completedTask == task)
-                .BecauseOf(because, becauseArgs)
-                .FailWith("Expected {context:task} to complete within {0}{reason}.", timeSpan);
+        if (completedTask == task)
+        {
+            timeoutCancellationTokenSource.Cancel();
+            await completedTask;
+        }
+
+        Execute.Assertion
+            .ForCondition(completedTask == task)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context:task} to complete within {0}{reason}.", timeSpan);
+
+        return new AndConstraint<TAssertions>((TAssertions)this);
+    }
+
+    /// <summary>
+    /// Asserts that the current <typeparamref name="TTask"/> will complete within the specified time.
+    /// </summary>
+    /// <param name="timeSpan">The allowed time span for the operation.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public async Task<AndConstraint<TAssertions>> NotCompleteWithinAsync(
+        TimeSpan timeSpan, string because = "", params object[] becauseArgs)
+    {
+        bool success = Execute.Assertion
+            .ForCondition(Subject is not null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context:task} to not complete within {0}{reason}, but found <null>.", timeSpan);
+
+        if (!success)
+        {
+            // subject is null, nothing to execute
+            return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        ITimer timer = Clock.StartTimer();
+        TTask task = Subject.Invoke();
+        TimeSpan remainingTime = timeSpan - timer.Elapsed;
+        if (remainingTime < TimeSpan.Zero)
+        {
+            // expected timeout reached, done
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        using var timeoutCancellationTokenSource = new CancellationTokenSource();
+        Task completedTask =
+            await Task.WhenAny(task, Clock.DelayAsync(remainingTime, timeoutCancellationTokenSource.Token));
+
+        if (completedTask == task)
+        {
+            timeoutCancellationTokenSource.Cancel();
+            await completedTask;
+        }
+
+        Execute.Assertion
+            .ForCondition(completedTask != task)
```

Comment:
♻️ You can skip this condition.

Reply:
Sorry, but I do not understand why,
The `if` before only checks whether the operation needs to be cancelled.

Reply:
If you move it within the `if` block above, it would work as expected. But also check my rewritten version in the other comment.

Reply:
obsolete, now